website/docs/plugin/framework/handling-data/terraform-concepts.mdx

```diff
@@ -0,0 +1,729 @@
+                                        Optional: true,
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+
+            "set_nested_block": schema.SetNestedBlock{
+                NestedObject: schema.NestedBlockObject{
+                    Attributes: map[string]schema.Attribute{
+                        "map_attribute": schema.MapAttribute{
+                            Optional:    true,
+                            ElementType: types.StringType,
+                        },
+                        "number_attribute": schema.NumberAttribute{
+                            Optional: true,
+                        },
+                        "object_attribute": schema.ObjectAttribute{
+                            Optional: true,
+                            AttributeTypes: map[string]attr.Type{
+                                "bool_attribute":    types.BoolType,
+                                "float64_attribute": types.Float64Type,
+                                "int64_attribute":   types.Int64Type,
+                                "list_attribute":    types.ListType{ElemType: types.StringType},
+                                "map_attribute":     types.MapType{ElemType: types.StringType},
+                                "number_attribute":  types.NumberType,
+                                "set_attribute":     types.ListType{ElemType: types.StringType},
+                                "string_attribute":  types.StringType,
+                            },
+                        },
+                        "set_attribute": schema.SetAttribute{
+                            Optional:    true,
+                            ElementType: types.StringType,
+                        },
+                    },
+                    Blocks: map[string]schema.Block{
+                        "set_nested_nested_block": schema.SetNestedBlock{
+                            NestedObject: schema.NestedBlockObject{
+                                Attributes: map[string]schema.Attribute{
+                                    "bool_attribute": schema.BoolAttribute{
+                                        Optional: true,
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+
+            "single_nested_block": schema.SingleNestedBlock{
+                Attributes: map[string]schema.Attribute{
+                    "bool_attribute": schema.BoolAttribute{
+                        Optional: true,
+                    },
+                    "float64_attribute": schema.Float64Attribute{
+                        Optional: true,
+                    },
+                },
+            },
+        },
+    }
+}
+```
+
+## Custom Types
```

Comment:
For a followup change, since this is a framework-defined concept and not a Terraform one, I'd suggest moving this to its own new page. 👍